config/src/main/java/org/springframework/security/config/annotation/web/configurers/SessionManagementConfigurer.java

```diff
@@ -561,13 +561,6 @@ InvalidSessionStrategy getInvalidSessionStrategy() {
 			this.invalidSessionStrategy = new SimpleRedirectInvalidSessionStrategy(
 					this.invalidSessionUrl);
 		}
-		if (this.invalidSessionUrl == null) {
```

Comment:
@OrangeDog Thanks for looking into this.

Taking a look at `getExpiredSessionStrategy` below, it looks like there may be an opportunity for cleanup there as well.

What would you think of making these two methods look the same, something like:

```java
if (strategy != null) {
    return strategy;
}

if (url == null) {
    return null;
}

strategy = // get the default strategy
return strategy;
```

I believe this would get rid of the dead branches as well as keep each if statement simple.